### DISCUSSION TITLE
An independent elastoplasticity (CamClay) model for soil mechanics

### DISCUSSION THREAD

**Discussion by mahyarmalek** [idaholab/moose#23424] [Q&A Modules: Solid mechanics]
I am writing to inform you of my recent development of an AD constitutive material with applications in soil mechanics, specifically the CamClay model. After successfully testing this code in the C++ library Deal.II, I have decided to develop the AD version in MOOSE to add additional features without the need for derivatives and Jacobians.

Despite attempting to fit the model into different subclasses of MOOSE to minimize manipulation, my first attempt using the ADGeneralizedRadialReturnMapping was unsuccessful due to its tie to the elasticity tensor, which conflicts with CamClay's unique elasticity tensor update. As a result, I have tried directly implementing the CamClay model as an ADStrainIncrementBasedStressUpdate. However, although I have successfully compiled the code, it does not converge for unknown reasons. I am particularly confused by the interdependencies of parameters, such as _stress[_qp], _incremental_strain[_qp], and others that must be provided.

To diagnose the issue, I have run several tests and have concluded that the problem occurs in the first step of the analysis before entering the plastic region. I suspect that the inconsistency of the initial stress and strain may be at fault. To elaborate, the CamClay model begins with an initial confining pressure (e.g., -100kPa) and then applies a deviatoric stress on top of a biaxial plain strain sample from the top.

If you have any recommendations or advice on how to resolve this issue or any potential oversights, please do not hesitate to let me know. I appreciate any help and am eager to resolve this issue as soon as possible.

Mahyar

**Comment by Rahim-Habibi** [comment]
Does initial stress distribute in the model properly?  can you paste here the initial stress state.
